### ISSUE TITLE
Extension for blocking snuff sites?

### ISSUE THREAD

**Issue by paxperscientiam** [StevenBlack/hosts#404]
Hello. Is there currently an extension for blocking snuff sites? It doesn't seem so.

**Comment by ScriptTiger** [comment]
Do you mean violence/excessive violence?

**Comment by paxperscientiam** [comment]
Yes.

**Comment by ScriptTiger** [comment]
Most people may only relate the word "porn" to material with graphic nudity and sexual content, but technically the actual definition would cover any obscenities such as this. I am sure if you have some particular suggestions in mind, you can open issues or make PRs with the porn sources:
https://github.com/Sinfonietta/hostfiles
https://github.com/Clefspeare13/pornhosts
What is everyone else's take on this? @paxperscientiam, can you list some examples of sites you are talking about just so we have some references for further discussion on this?

**Comment by paxperscientiam** [comment]
@ScriptTiger Suggestions? Oh, boy...here are some examples:

bitcheese.net
horriblevideos.com
deathaddict.com

**Comment by ScriptTiger** [comment]
The only worry I have is if we start expanding categories in this manner, it could grow to be unmanageable. If we include one for violence, then maybe it could be an opening to also include one for drug usage, terrorist-related content, and other material supporting legally-controversial activities. If we change to a rating-based system similar to how the MPAA rates films, it would be putting a lot of responsibility on the curators to uphold a more strict standard across the multitude of domains they have in their lists. I also don't really like including legally-controversial activity with porn because, as much as some parents may think otherwise, porn is not actually illegal in most countries. There's also the issue with even grouping extreme "illegal" material together because things like marijuana usage or homosexuality get the death penalty in some countries while are totally legal in others. And there's things like discerning a hate message of violence from a possible educational or news source. However, if we can all simply agree that websites containing purely violence with no clear positive message whatsoever, are bad, that seems like a broad enough topic to both keep the project lightweight and let the sources more easily collect their lists.

Whatever our decision, we should consider the goal of this project is ultimately to keep a lightweight list and not to go too overboard in any one direction. I think this is definitely a good idea to look into though and it's definitely another case for documentation outlining the standards and expectation of this project as a whole and also for the individual extensions.

**Comment by StevenBlack** [comment]
There are two facets here.

The first facet is the number of combinations as we increase the number of extensions.  We need a way to manage this programmatically.  Currently we [do this explicitly](https://github.com/StevenBlack/hosts/blob/5f2c5a1bf720eb961c136a61ec33cceb526837dd/makeHosts.py#L56-L114).

The second facet is the overlap between facets gets increasingly nebulous.

For example `4chan` is currently included in the porn extension sources.  4chan has everything.

At the moment, to keep things manageable, I'm inclined to bundle this under porn.

**Comment by ScriptTiger** [comment]
How do @Clefspeare13 and @Sinfonietta source their data? Would adding violence to the mix be an easy thing for them? If not, maybe we can just add a local porn/stevenblack/hosts temporarily to take PRs for this while we find a more dedicated source. Either way, while we're using the porn extension whoever ends up sourcing it should at least make a new commented section in the porn hosts to keep them together just in case we think differently in the future.

**Comment by StevenBlack** [comment]
I think a Pull Request with @Clefspeare13 and @Sinfonietta would work well...

**Comment by Clefspeare13** [comment]
I would be willing to include this in my sources. I would also be able to create a host file that includes violence and one that does not, though that may complicate things more for Steven.

**Comment by rautamiekka** [comment]
I've been thinking about a system which'd unite this big, manual work and profile info on the stuff in the lists (who blocked it why, etc.), but all the otherwise good ideas don't really make it easier, making my head smoke.

**Comment by paxperscientiam** [comment]
@StevenBlack 

> Currently we do this explicitly.

Perhaps the program could be made to read from a config that specifies extensions in terms of Github repo names?

EG:
````JSON
{
    "extensions": [
        "marktron/fakenews",
        "Sinfonietta/hostfiles"
    ]
}
````

> The second facet is the overlap between extensions gets increasingly nebulous.

I'm not so sure this is a big problem. For one, any measure of nebulosity depends on an agreed upon standard on what constitutes "fake news"and "pornography." That said, blocking trackers is not a nebulous concept.

Moreover, the end-user is always free to white-list any given domain. That said, you'll still have to way how comfortable you are with the fact that most end-users will not be checking each domain for accuracy. That said, some level of tech-savvy can be safely assumed given the nature of the project.

In short, since 4chan has violent material, pornographic material, and fake news, it, practically speaking, doesn't matter which list it's on.

I assume duplicates are stripped, yes? If attribution is a concern, perhaps keep the first occurrence of "0.0.0.0 sexyviolentfakenews.net" and comment out subsequent occurrences (as opposed to stripping).

@ScriptTiger 
> ...keep a lightweight list and not to go too overboard in any one direction

I agree; that's why extensions are important.


I'll make a PR.

**Comment by StevenBlack** [comment]
@paxperscientiam I like your config idea.

**Comment by paxperscientiam** [comment]
@Clefspeare13 Hey, I overlooked your post. I've since submitted a PR, though I was unsure of how you compiled things. I wound up editing both the 0s and 127's .
